### PR TITLE
Testing/2.7.2

### PR DESCRIPTION
**Pull request by jmarrec**
Bump to 2.7.2. Note: this requires the creation of the testing/2.7.2 branch

### CODE REVIEW

**Comment by ericLemanissier** [comment]
trigger CI

**Comment by jmarrec** [comment]
So I had to add openssl to the mix, because even if you pass `--with-out-ext=openssl` it appears to be an unresolved dependency on unix, and so the build fails.

Not sure if I shouldn't pass the path to the conan openssl to make sure that it picks up the one we specify and not the system one.  I added that in the last commit c91668656008830df6e2651b8ba33152c81736de, which can be reverted if you do not like it.

File: conanfile.py [hunk 1]
```diff
@@ -32,9 +32,11 @@ def requirements(self):
     def build_requirements(self):
         if self.settings.os_build == "Windows":
             self.build_requires("7zip/19.00")
+        else:
+            self.build_requires("openssl/1.1.1h")
```
Comment (jmarrec) [inline conanfile.py:36]: Need openssl or 

```
/Users/runner/.conan/data/ruby_installer/2.7.2/bincrafters/testing/build/743cf0321be3152777da4d05247a66d1552e70a2/source_subfolder/lib/rubygems/core_ext/kernel_require.rb:92:in `require': cannot load such file -- openssl (LoadError)
```

File: conanfile.py [hunk 2]
```diff
@@ -48,9 +50,11 @@ def _configure_autotools(self):
             self._autotools = AutoToolsBuildEnvironment(self)
             args = [
                 "--disable-install-doc",
-                "--with-out-ext=gdbm,openssl,pty,readline,syslog",
+                "--with-out-ext=gdbm,pty,readline,syslog",
```
Comment (jmarrec) [inline conanfile.py:53]: Enable openssl, as it appears needed after all.

File: conanfile.py [hunk 3]
```diff
@@ -48,9 +50,11 @@ def _configure_autotools(self):
...
                 "--without-gmp",
-                "--enable-load-relative"
+                "--enable-load-relative",
+                "--with-openssl-dir={}".format(self.deps_cpp_info["openssl"]
+                                                   .rootpath)
```
Comment (jmarrec) [inline conanfile.py:57]: Optional, can be reverted.

Specify --with-openssl-dir to conan openssl to ensure it's the one picked up.

**Review by Croydon — APPROVED** [review]

**Comment by Croydon** [comment]
I'm not sure why the CI in the PR succeeded but the branch push fails... https://github.com/bincrafters/conan-ruby_installer/actions/runs/368026105

Could you have a look @jmarrec?

**Comment by ericLemanissier** [comment]
it could be because the PR used docker image `conanio/gcc7` whereas the push uses `conanio/gcc7-centos6`

**Comment by jmarrec** [comment]
It does seem to be an openssl-related problem.

```
make[2]: Entering directory '/home/conan/.conan/data/ruby_installer/2.7.2/bincrafters/testing/build/925368c87e3030fc930c24f33d45f66d5f7abbfb/ext/openssl'
compiling ../../source_subfolder/ext/openssl/openssl_missing.c
make[2]: Leaving directory '/home/conan/.conan/data/ruby_installer/2.7.2/bincrafters/testing/build/925368c87e3030fc930c24f33d45f66d5f7abbfb/ext/rbconfig/sizeof'
make[2]: Entering directory '/home/conan/.conan/data/ruby_installer/2.7.2/bincrafters/testing/build/925368c87e3030fc930c24f33d45f66d5f7abbfb/ext/coverage'
compiling ../../source_subfolder/ext/coverage/coverage.c
In file included from ../../source_subfolder/ext/openssl/openssl_missing.c:21:0:
../../source_subfolder/ext/openssl/openssl_missing.h:75:0: warning: "X509_STORE_get_ex_new_index" redefined
 #  define X509_STORE_get_ex_new_index(l, p, newf, dupf, freef) \
 
In file included from /home/conan/.conan/data/openssl/1.1.1h/_/_/package/d50a0d523d98c15bb147b18fa7d203887c38be8b/include/openssl/x509.h:320:0,
                 from /home/conan/.conan/data/openssl/1.1.1h/_/_/package/d50a0d523d98c15bb147b18fa7d203887c38be8b/include/openssl/pem.h:17,
                 from /home/conan/.conan/data/openssl/1.1.1h/_/_/package/d50a0d523d98c15bb147b18fa7d203887c38be8b/include/openssl/ui.h:19,
                 from /home/conan/.conan/data/openssl/1.1.1h/_/_/package/d50a0d523d98c15bb147b18fa7d203887c38be8b/include/openssl/engine.h:24,
                 from ../../source_subfolder/ext/openssl/openssl_missing.c:14:
/home/conan/.conan/data/openssl/1.1.1h/_/_/package/d50a0d523d98c15bb147b18fa7d203887c38be8b/include/openssl/x509_vfy.h:330:0: note: this is the location of the previous definition
 #define X509_STORE_get_ex_new_index(l, p, newf, dupf, freef) \
 
In file included from /home/conan/.conan/data/openssl/1.1.1h/_/_/package/d50a0d523d98c15bb147b18fa7d203887c38be8b/include/openssl/bn.h:20:0,
                 from /home/conan/.conan/data/openssl/1.1.1h/_/_/package/d50a0d523d98c15bb147b18fa7d203887c38be8b/include/openssl/engine.h:18,
                 from ../../source_subfolder/ext/openssl/openssl_missing.c:14:
../../source_subfolder/ext/openssl/openssl_missing.c: In function 'ossl_HMAC_CTX_new':
../../source_subfolder/ext/openssl/openssl_missing.c:65:43: error: invalid application of 'sizeof' to incomplete type 'HMAC_CTX {aka struct hmac_ctx_st}'
     HMAC_CTX *ctx = OPENSSL_malloc(sizeof(HMAC_CTX));
                                           ^
/home/conan/.conan/data/openssl/1.1.1h/_/_/package/d50a0d523d98c15bb147b18fa7d203887c38be8b/include/openssl/crypto.h:118:23: note: in definition of macro 'OPENSSL_malloc'
         CRYPTO_malloc(num, OPENSSL_FILE, OPENSSL_LINE)
                       ^~~
../../source_subfolder/ext/openssl/openssl_missing.c:68:5: warning: implicit declaration of function 'HMAC_CTX_init'; did you mean 'HMAC_CTX_new'? [-Wimplicit-function-declaration]
     HMAC_CTX_init(ctx);
     ^~~~~~~~~~~~~
     HMAC_CTX_new
../../source_subfolder/ext/openssl/openssl_missing.c: In function 'ossl_HMAC_CTX_free':
../../source_subfolder/ext/openssl/openssl_missing.c:78:2: warning: implicit declaration of function 'HMAC_CTX_cleanup'; did you mean 'HMAC_CTX_get_md'? [-Wimplicit-function-declaration]
  HMAC_CTX_cleanup(ctx);
  ^~~~~~~~~~~~~~~~
  HMAC_CTX_get_md
../../source_subfolder/ext/openssl/openssl_missing.c: In function 'ossl_X509_CRL_get0_signature':
../../source_subfolder/ext/openssl/openssl_missing.c:90:13: error: dereferencing pointer to incomplete type 'X509_CRL {aka const struct X509_crl_st}'
  *psig = crl->signature;
             ^~
../../source_subfolder/ext/openssl/openssl_missing.c: In function 'ossl_X509_REQ_get0_signature':
../../source_subfolder/ext/openssl/openssl_missing.c:102:13: error: dereferencing pointer to incomplete type 'X509_REQ {aka const struct X509_req_st}'
  *psig = req->signature;
             ^~
make[2]: *** [Makefile:313: openssl_missing.o] Error 1
make[2]: Leaving directory '/home/conan/.conan/data/ruby_installer/2.7.2/bincrafters/testing/build/925368c87e3030fc930c24f33d45f66d5f7abbfb/ext/openssl'
make[1]: *** [exts.mk:257: ext/openssl/all] Error 2
make[1]: *** Waiting for unfinished jobs....
installing default coverage libraries
linking shared-object coverage.so
make[2]: Leaving directory '/home/conan/.conan/data/ruby_installer/2.7.2/bincrafters/testing/build/925368c87e3030fc930c24f33d45f66d5f7abbfb/ext/coverage'
make[1]: Leaving directory '/home/conan/.conan/data/ruby_installer/2.7.2/bincrafters/testing/build/925368c87e3030fc930c24f33d45f66d5f7abbfb'
make: *** [uncommon.mk:295: build-ext] Error 2
```

**Pull request merged**